File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
input = sys.stdin.readline

n, m, v = map(int, input().split())
graph = [[0 for _ in range(n+1)] for _ in range(n+1)]

for _ in range(m):
    a, b = map(int, input().split())
    graph[a][b] = 1
    graph[b][a] = 1


def dfs():
    visited = []

    def recursive(cur):
        if cur not in visited: visited.append(cur)

        for i in range(1, n+1):
            if graph[cur][i] == 1 and i not in visited:
                recursive(i)

    recursive(v)
    print(' '.join(map(str, visited)))

def bfs():
    visited = []
    queue = []
    queue.append(v)


    while(queue):
        cur = queue.pop(0)
        if cur not in visited: visited.append(cur)

        for i in range(1, n+1):
            if graph[cur][i] == 1 and i not in visited:
                queue.append(i)
     
    print(' '.join(map(str, visited)))

dfs()
bfs()

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    static int n, m, v;
    static int[][] graph;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());

        graph = new int[n + 1][n + 1];

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            graph[a][b] = 1;
            graph[b][a] = 1;
        }

        visit = new boolean[n + 1];
        dfs(v);
        System.out.println();
        bfs();

    }

    static boolean[] visit;

    static void dfs(int cur) {
        if (!visit[cur]) {
            visit[cur] = true;
            System.out.print(cur + " ");
        }

        for (int i = 1; i <= n; i++) {
            if (graph[cur][i] == 1 && !visit[i]) dfs(i);
        }
    }

    static void bfs() {
        visit = new boolean[n + 1];
        Queue<Integer> q = new LinkedList<>();
        q.offer(v);

        while (!q.isEmpty()) {
            int cur = q.poll();
            if (!visit[cur]) {
                visit[cur] = true;
                System.out.print(cur + " ");
            }

            for (int i = 1; i <= n; i++) {
                if (graph[cur][i] == 1 && !visit[i]) q.offer(i);
            }
        }
    }
}